File: swift/DesignPattern.playground/Pages/FactoryMethodPattern.xcplaygroundpage/Sources/USBeverageFactory.swift
import Foundation

open class USBeverageFactory: BeverageFactory {
    
    public init() {}
    
    public func createBeverage(beverageName: String) -> Beverage? {
        var beverage: Beverage?
        
        switch beverageName {
        case "black tea":
            beverage = CeylonBlackTea()
        case "green tea":
            beverage = GyokuroGreenTea()
        default:
            break
        }
        
        return beverage
    }
}

File: kotlin/DesignPattern/src/main/kotlin/factorymethodpattern/source/USBeverageFactory.kt
package factorymethodpattern.source

class USBeverageFactory: BeverageFactory {
    override fun createBeverage(beverageName: String): Beverage? {
        return when (beverageName) {
            "black tea" -> CeylonBlackTea()
            "green tea" -> GyokuroGreenTea()
            else -> null
        }
    }
}